core/src/main/scala/kafka/zk/AdminZkClient.scala

```diff
@@ -153,16 +157,17 @@ class AdminZkClient(zkClient: KafkaZkClient) extends Logging {
     LogConfig.validate(config)
   }
 
-  private def writeTopicPartitionAssignment(topic: String, replicaAssignment: Map[Int, ReplicaAssignment], isUpdate: Boolean): Unit = {
+  private def writeTopicPartitionAssignment(topic: String, replicaAssignment: Map[Int, ReplicaAssignment],
+                                            isUpdate: Boolean, usesTopicId: Boolean = false): Unit = {
```

Comment:
Why do you set the default value to false? I don't think we should give it a default value since the caller can decide it by caller.

Reply:
Some places that eventually call this do not have access to IBP. The controller will eventually pick up and assign topic IDs that need them.